### ISSUE TITLE
Segfault on linux

### ISSUE THREAD

**Issue by sbruder** [ppy/osu-framework#3058]
### Environment

I build and run osu!lazer in my own docker container. You can find the Dockerfile here: https://git.sbruder.de/docker/osu/src/branch/master/Dockerfile
The command I use to run osu!lazer is:
```
docker run \
	--rm \
	-it \
	--name osu \
	-e DISPLAY \
	-e PULSE_SERVER=/run/user/$(id -u)/pulse/native \
	-e PULSE_COOKIE=/run/pulse/cookie \
	--device /dev/dri \
	--group-add $(getent group video|cut -d: -f3) \
	--group-add $(getent group audio|cut -d: -f3) \
	-v /tmp/.X11-unix:/tmp/.X11-unix \
	-v /run/user/$(id -u)/pulse/native:/run/user/$(id -u)/pulse/native \
	-v $HOME/.config/pulse/cookie:/run/pulse/cookie \
	-v $HOME/opt/osu/data/:/home/osu/osu/ \
	osu-lazer \
	$@
```

Up to osu!lazer commit https://github.com/ppy/osu/commit/47d0eb1564fab13184d3ba5ca231d9c1dd8492a8 this has been working.

### Current behaviour
Starting with commit https://github.com/ppy/osu/commit/b9a8a36e60e633e94042734d2a7dc8b81b74bfe3 (which updates the osu framework to version 2019.1128.1) the container exits with return code 139 (segmentation fault).
Since the only thing changed in the commit is updating the framework version, I decided to open the issue in this repository. I hope this is the right place.

### Steps to reproduce

Build the linked docker file with the `GIT_COMMIT` build argument set to any commit set to any commit after https://github.com/ppy/osu/commit/47d0eb1564fab13184d3ba5ca231d9c1dd8492a8 or without any build argument (uses the latest master commit):

    docker build --build-arg GIT_COMMIT=b9a8a36e60e633e94042734d2a7dc8b81b74bfe3 -t osu-lazer .

As I do not have a .NET core environment set up, I cannot test if this also happens outside of the docker container.

The error seems to be introduced between the commits 1067c305162c2e039ea770fcd0c7ef781677497a and 198f01e4e42ad5c34fd186124a6fe7aa170f49bd.

**Comment by bdach** [comment]
Did you try bisecting that commit range? It shouldn't be too taxing if the error reproduces in visual tests. `dmesg` output post-segfault would also be nice.

**Comment by sbruder** [comment]
Thank you for your reply.

This is the output of dmesg related to the segfault:
```
[114810.760505] dotnet[320798]: segfault at 8 ip 00007fbf885f35d9 sp 00007fbfb8ff5160 error 4 in libbass_fx.so[7fbf885e2000+16000]
[114810.760511] Code: 3d 4c 52 20 00 e9 e7 1d ff ff 48 83 ec 08 83 3d 34 52 20 00 00 75 16 48 8b 05 23 52 20 00 be 03 00 00 00 48 8b 3d d7 49 20 00 <ff> 50 08 5e e9 a8 ff ff ff 90 90 90 90 90 90 90 90 90 90 90 90 90
```

Unfortunately I do not have much experience with .NET, so I don’t really know how to build osu!lazer with a specific commit of osu-framework. I tried to look into the (thankfully few) changes between the two versions and the only commits that add changes that were not reverted again are 21d604837f4abaa33f00c13eb819e074b2957047 and b96cdc54507c3cd161f867ccb705771f9474e538.

**Comment by bdach** [comment]
Looks like it might be the same thing as ppy/osu#7018, which was closed with #3057. Would you mind trying to `dotnet run osu.Framework.Tests` from this repository's latest `master` vs 3bd14c80 and seeing if it resolves the problem?

(Alternatively, see [instructions for how to run lazer with another framework version](https://github.com/ppy/osu-framework/wiki/Testing-local-framework-checkout-with-other-projects) - requires having a clone of both and `jq` installed)

**Comment by sbruder** [comment]
Thank you very much. Sorry for not seeing that issue. The pull request fixes the issue.
